src/androidTest/java/com/couchbase/lite/support/BatcherTest.java

```diff
@@ -285,6 +282,14 @@ public void process(List<String> itemsToProcess) {
             success = latch1.await(500, TimeUnit.MILLISECONDS);
             assertTrue(success);
 
+            // NOTE: https://github.com/couchbase/couchbase-lite-java-core/issues/518
+            //       CountDownLatch became 0, but task might not be done yet.
+            //       Give little bit of time for slower environment.
+            try {
+                Thread.sleep(500);
+            } catch (Exception ex) {
+            }
+
             // add another object
             batcher.queueObject(new String());
```

Comment:
https://github.com/couchbase/couchbase-lite-java-core/blob/b37a40b4786292246aa81c3517b531de0dc83c3b/src/main/java/com/couchbase/lite/support/Batcher.java#L215-L222

When the latch1 became 0 and a new string added to the batcher queue, the inbox size will be more than 0 and the code above will be executed. This could happen in slow environment. I think you can simulate the behavior by adding some sleep right before line 215 in Batcher.java.

If above happens, the new string added to the queue will get cancelled here:
https://github.com/couchbase/couchbase-lite-java-core/blob/b37a40b4786292246aa81c3517b531de0dc83c3b/src/main/java/com/couchbase/lite/support/Batcher.java#L93-L101